### DISCUSSION TITLE
FsLab nuget package

### DISCUSSION THREAD

**Discussion by kMutagene** [fslaborg/FsLab#9] [General]
I have thought a lot about integrating the fslab stack more tightly recently. 

The next logical step for that would be composite packages offering e.g.
- Plotly.NET chart APIs directly taking Deedle frames as input (a bit like plotly express for python,. which works on pandas frames)
- FSharp.Stats functions taking deedle frames as input (e.g. functions that convert Deedle data to matrix, perform calculations, and return frames again)
- Plotly.NET chart APIs functions for FSharp.Stats types e.g. for Distributions, PCA results etc.

One roadblock for such a package is that there is already an [existing fslab package on nuget](https://www.nuget.org/packages/FsLab). It was last updated 2018, but there still seem to be [people that use it today](https://www.nuget.org/stats/packages/FsLab?groupby=Version)

Would it be acceptable to expect those users to pin their versions and use that namespace for the composite package? 

Comparable ecosystems such as scipy or tidyverse can both be installed via their respective names:

```bash
python -m pip install scipy
conda install scipy
```

```R
install.packages("tidyverse")
```

And i think having that for F#/.NET would be a great thing

```fsharp
#r "nuget: FsLab"
// contains the core fslab stack as well as the notebook extensions for those and composite functions
```

As original owners of that package, what are your thoughts on this @dsyme @tpetricek ?

**Comment by tpetricek** [comment]
I think having a package "FsLab version 2" that is not backwards compatible (and contains stuff that is actually useful today) would be amazing. It is definitely reasonable for existing users of FsLab to pin their package version to "1.*" and not expect it to be updated (because it contains packages that themselves have not been updated in a long time).

If "FsLab v2" also worked directly in .NET Interactive notebooks (with all the necessary visualizers installed), that would be very nice.

**Comment by dsyme** [comment]
Yes totally reasonable to expect users to pin, and indeed best to allow yourself as many non-compat major version upgrades as you like in the future as well.

**Comment by kMutagene** [comment]
Okay awesome! Can one of you add me as owner of the nuget package so that's already taken care of once i get around to work on this?

**Comment by dsyme** [comment]
@kMutagene Added, pending your approval!
